---
source: crates/uitest/tests/ty.rs
expression: diags
input_file: crates/uitest/fixtures/ty/kind_bound.fe
---
error[2-0002]: `wrapper3` is not found
   ┌─ kind_bound.fe:34:15
   │
34 │     bar_err3: wrapper3<Wrapper1, Wrapper1>,
   │               ^^^^^^^^ `wrapper3` is not found

error[3-0001]: invalid type argument kind
   ┌─ kind_bound.fe:25:14
   │
25 │     foo_err: Wrapper2<i32, i32>,
   │              ^^^^^^^^^^^^^^^^^^ expected `(* -> *)` kind, but `i32` has `*` kind

error[3-0001]: invalid type argument kind
   ┌─ kind_bound.fe:32:15
   │
32 │     bar_err1: Wrapper3<i32, u32>,
   │               ^^^^^^^^^^^^^^^^^^ expected `((* -> *) -> (* -> *))` kind, but `i32` has `*` kind

error[3-0001]: invalid type argument kind
   ┌─ kind_bound.fe:33:15
   │
33 │     bar_err2: Wrapper3<Wrapper2, u32>,
   │               ^^^^^^^^^^^^^^^^^^^^^^^ expected `(* -> *)` kind, but `u32` has `*` kind

error[3-0005]: duplicate type bound is not allowed.
   ┌─ kind_bound.fe:38:10
   │
38 │ where T: (* -> *) -> *
   │          ^^^^^^^^^^^^^ `T` is already declared with `(* -> *)` kind, but found `((* -> *) -> *)` kind here

error[6-0005]: trait bound for concrete type is not allowed
   ┌─ kind_bound.fe:44:7
   │
44 │ where Self: *
   │       ^^^^ `InvalidBound2` is a concrete type

error[6-0005]: trait bound for concrete type is not allowed
   ┌─ kind_bound.fe:50:7
   │
50 │ where Wrapper1<i32>: *
   │       ^^^^^^^^^^^^^ `Wrapper1<i32>` is a concrete type


